skunkworks_crow/src/main/java/org/odk/share/views/ui/settings/SettingsActivity.java

```diff
@@ -29,10 +33,12 @@
 
 public class SettingsActivity extends PreferenceActivity {
 
+    private static final int DIRECTORY_REQUEST_CODE = 9999;
     EditTextPreference hotspotNamePreference;
     Preference hotspotPasswordPreference;
-    CheckBoxPreference passwordRequirePreference;
-    EditTextPreference odkDestinationDirPreference;
+    SwitchPreference passwordRequirePreference;
+    Preference odkDestinationDirPreferenceDirectoryPicker;
```

Comment:
Creating two different preferences is fine, but the values that we get from these two different preferences on different versions stored separately. Ideally, whatever value that we receive from these UI Objects should be saved to the same preference.

Reply:
I tried that but I am getting preference cast exception

Reply:
I'll give it a shot then will suggest the solution

Reply:
Yeah than you